packages/wrangler/src/api/startDevWorker/ConfigController.ts

```diff
@@ -53,14 +53,16 @@ async function resolveDevConfig(
 	config: Config,
 	input: StartDevWorkerInput
 ): Promise<StartDevWorkerOptions["dev"]> {
-	const auth =
-		input.dev?.auth ??
-		(async () => {
-			return {
-				accountId: await getAccountId(),
-				apiToken: requireApiToken(),
-			};
-		});
+	const auth = async () => {
+		if (input.dev?.auth) {
+			return unwrapHook(input.dev.auth, config);
```

Comment:
I'm not sure this is the same behaviour. This seems to unconditionally unwrap the hook if provided, even if not in remote mode?

Reply:
It doesn't unconditionally unwrap the hook. It unconditionally sets the auth property with its own hook which will unwrap the original property when itself is unwrapped

(This diff answers your other question. https://github.com/cloudflare/workers-sdk/pull/6963/files#r1799613869)